core/src/main/java/greencity/webcontroller/ManagementRatingStatisticsController.java

```diff
@@ -80,4 +90,45 @@ public void exportToExcel(HttpServletResponse response) throws IOException {
         List<RatingStatisticsDto> ratingStatisticsList = ratingStatisticsService.getAllRatingStatistics();
         ratingExcelExporter.export(response.getOutputStream(), ratingStatisticsList);
     }
+
+    /**
+     * Export filtered {@link RatingStatistics} to Excel file.
+     *
+     * @author Dovganyuk Taras
+     */
+    @GetMapping("/exportFiltered")
+    public void exportFilteredToExcel(HttpServletResponse response) throws IOException {
+        response.setContentType("application/octet-stream");
+        String headerKey = "Content-Disposition";
+
+        String currentDate = dateFormat.format(new Date());
+        String fileName = "user_rating_statistics" + currentDate + ".xlsx";
+        String headerValue = "attachment; filename=" + fileName;
+
+        response.setHeader(headerKey, headerValue);
+
+        List<RatingStatisticsDto> ratingStatisticsList =
+            ratingStatisticsService.getFilteredRatingStatisticsForExcel(spec);
+        ratingExcelExporter.export(response.getOutputStream(), ratingStatisticsList);
+    }
+
+    /**
+     * Returns  management page with User rating statistics with filtered data.
+     *
+     * @param model                   ModelAndView that will be configured and returned to user.
+     * @param ratingStatisticsViewDto used for receive parameters for filters from UI.
+     */
+    @PostMapping(value = "", consumes = MediaType.APPLICATION_FORM_URLENCODED_VALUE)
+    public String filterData(Model model,
+                             @PageableDefault(value = 20) @ApiIgnore Pageable pageable,
+                             RatingStatisticsViewDto ratingStatisticsViewDto) {
+        SearchCriteria searchCriteria = ratingStatisticsService.buildSearchCriteria(ratingStatisticsViewDto);
+        Pageable paging =
+            PageRequest.of(pageable.getPageNumber(), pageable.getPageSize(), Sort.by("createDate").descending());
+        spec.setSearchCriteria(searchCriteria);
```

Comment:
This is awful. Imagine 10 users making request simultaneously. Whos `searchCriteria` will be set into the `spec`?